modulo7/Cookenu/src/controller/UserController.ts

```diff
@@ -0,0 +1,93 @@
+import { Request, Response } from "express";
+import { UserBusiness } from "../business/UserBusiness";
+import { Unauthorized } from "../error/customError";
+import { LoginInputDTO, UserInputDTO } from "../model/user";
+import { Authenticator } from "../services/Authenticator";
+
+export class UserController {
+
+  public signup = async (req: Request, res: Response) => {
+    try {
+      const { name, email, password, role } = req.body;
+
+      const input: UserInputDTO = {
+        name,
+        email,
+        password,
+        role
+      };
+      const userBusiness = new UserBusiness()
+      const token = await userBusiness.signup(input);
+
+      res.status(201).send({ message: "Usuário criado!", token });
+    } catch (error: any) {
+      res.status(400).send(error.message);
```

Comment:
Passando o catch dessa forma os erros vão acabar sendo sempre 400. Dê uma olhada em custom errors e o instance of!